pkg/arvo/app/chat-store.hoon

```diff
@@ -197,8 +232,11 @@
     [~ state]
   =.  letter.envelope.action  (evaluate-letter [author letter]:envelope.action)
   =^  envelope  u.mailbox  (prepend-envelope u.mailbox envelope.action)
-  :-  (send-diff path.action action(envelope envelope))
-  state(inbox (~(put by inbox) path.action u.mailbox))
+  :_  state(inbox (~(put by inbox) path.action u.mailbox))
+  ?:  =((mod number.envelope 5.000) 0)
+    :-  [%pass /trim %agent [our.bol %chat-store] %poke %noun !>([%trim ~])]
+    (send-diff path.action action(envelope envelope))
+  (send-diff path.action action(envelope envelope))
```

Comment:
idk why we always defer this to new events instead of just running them in-place. What's the reasoning there?

Also every 5.000 messages seems very infrequent, especially per channel. I'd put this at more like 500 or something. But also, considering `+handle-messages` exists, you could skip some of these checks, especially in cases where you're, you know, resubscribing.

Reply:
5.000 seems like a good frequency. 5.000 messages really don't take up that much space.

Reply:
As for deferring to new events, I think it's a cleaner pattern for this type of operation.